tool/tctl/common/token_command.go

```diff
@@ -528,7 +579,488 @@ func (c *TokensCommand) List(ctx context.Context, client *authclient.Client) err
+	}
+
+	proxyAddr := authPong.GetProxyPublicAddr()
+	if proxyAddr == "" {
+		return trace.BadParameter("failed to discover Teleport proxy address, make sure the Teleport Proxy service is running with `public_addr` set")
+	}
+	agentVersion := proxyPong.AutoUpdate.AgentVersion
+	if agentVersion == "" {
+		return trace.NotFound("failed to discover Teleport Agent version")
+	}
+
+	fmt.Fprintln(os.Stderr, "📁 Finding local kubectl context")
+	kubeContext := c.kubeContext
+
+	// If kube context not set, we look it up
+	if kubeContext == "" {
+		kubeContext, err = lookupKubeActiveContext(kubectlPath)
+		if err != nil {
+			return trace.Wrap(err, "looking up active kubectl context")
+		}
+	}
+
+	// If the kube name is not specified, we use the context name
+	kubeName := c.kubeName
+	if kubeName == "" {
+		kubeName = kubeContext
+	}
+
+	// If the token name is not specified, we use the kube name suffixed by the bot name.
+	tokenName := c.tokenName
+	if tokenName == "" {
+		tokenName = kubeName
+		if c.botName != "" {
+			tokenName = tokenName + "-" + c.botName
+		}
+	}
+
+	tokenParams := createKubeTokenParams{
+		kubeContext:  kubeContext,
+		kubeName:     kubeName,
+		kubeJoinType: kubeJoinType,
+		kubectlPath:  kubectlPath,
+
+		roles:          roles,
+		botName:        c.botName,
+		namespace:      c.namespace,
+		serviceAccount: c.serviceAccountName,
+		tokenName:      tokenName,
+		insecure:       client.Config().InsecureSkipVerify,
+		force:          c.force,
+	}
+
+	token, err := createKubeToken(ctx, client, tokenParams)
+	if err != nil {
+		return trace.Wrap(err, "creating token")
+	}
+
+	chartName := "teleport/teleport-kube-agent"
+	if c.botName != "" {
+		chartName = "teleport/tbot"
+	}
+
+	fmt.Fprintf(os.Stderr, "📝 Writing %s Helm values to: %q\n", chartName, c.outputPath)
+
+	var valueGenerator valueGeneratorFunc
+	if c.botName != "" {
+		valueGenerator = generateTbotValues
+	} else {
+		valueGenerator = generateAgentValues
+	}
+
+	values, err := valueGenerator(valueGeneratorParams{
+		botName:             c.botName,
+		roles:               roles,
+		proxyAddr:           proxyAddr,
+		enterprise:          proxyPong.Edition == modules.BuildEnterprise,
+		updateGroup:         c.updateGroup,
+		tokenName:           token.GetName(),
+		teleportClusterName: authPong.GetClusterName(),
+		kubeClusterName:     kubeName,
+	})
+	if err != nil {
+		return trace.Wrap(err, "error generating chart values")
+	}
+
+	if err := os.WriteFile(c.outputPath, values, 0644); err != nil {
+		return trace.Wrap(err, "error writing chart values to file %s", c.outputPath)
+	}
+
+	fmt.Fprintln(os.Stderr, "🚀 You can now deploy the Helm chart:")
+	os.Stderr.Write([]byte("\n"))
+
+	fmt.Println(craftHelmCommand(craftHelmCommandParams{
+		releaseName: c.serviceAccountName,
+		chartName:   chartName,
+		namespace:   c.namespace,
+		version:     proxyPong.AutoUpdate.AgentVersion,
+		valuesPath:  c.outputPath,
+		kubeContext: kubeContext,
+	}))
```

Comment:
I think it's fine to just target Helm use here, but there's also a lot of non-helm tbot deployments (e.g. I'm currently writing up docs that use a tbot sidecar for Ansible AWX). What do you think about adding a mode for non-Helm use?

I figure that could be as small as a flag to skip writing Helm values, printing the audience value to use, or maybe projected token volume example, e.g.:
```yaml
  volumes:
    - name: join-sa-token
      projected:
        sources:
          - serviceAccountToken:
              path: join-sa-token
              expirationSeconds: 600
              audience: example.teleport.sh
```

Reply:
I think we'll be able to reuse the --format for this. If we need this I think we should do it in a separate PR as this one is already a monster